### ISSUE TITLE
Image -> input

### ISSUE THREAD

**Issue by kevin-shannon** [kevin-shannon/sudoku#9]
Ok this idea is a bit more out there, would take quite a bit of effort. But it would be cool to be able to open up a sudoku book, take a picture of a puzzle, and use computer vision to automatically fill in the input like for depositing a check online.

**Comment by tannerkrewson** [comment]
The tools definitely exist to attempt this. Not sure if it would work well in practice, but definitely worth attempting if you're interested. Here are the steps you would likely take:

1. See if you can open up the development Sudoku on your phone. Find the ip of your computer by using `ipconfig` and open `192.168.1.xxx:3000` in your phone browser.
2. Show view finder and take photo with [react-webcam](https://github.com/mozmorris/react-webcam)
3. Use [tesseract.js](https://github.com/naptha/tesseract.js) to get the numbers
4. Process the numbers and handle any errors
5. ???
6. It works maybe 🤷

**Comment by tannerkrewson** [comment]
actually........ i think it might be more difficult than i thought. i think the hardest part would be handling the blank spaces in a scanned board. not sure how tesseract would handle that... i'm not hopeful 🤔

**Comment by kevin-shannon** [comment]
maybe tensorflow.js with our own supervised learning model trained on normalized images of puzzles.

**Comment by tannerkrewson** [comment]
Yeah that would definitely work, just a lot more effort! But would be super
cool if you're down to hand transcribe a bunch of puzzles for training! 😄

**Comment by tannerkrewson** [comment]
_....unless?_

https://github.com/wichtounet/sudoku_dataset
